### PR TITLE
Several clean up updates to prepare for new branching rules

### PR DESCRIPTION
**Pull request by jamezp**
This PR cleans up the workflows slightly to ensure consistency. It also changes provisioning for each TCK runner which uses provisioning to use the `org.wildfly.plugins:wildfly-maven-plugin` for provisioning.

The README was updated with instructions on how we will handle branching. The branches will be used to test patch releases while allowing the main branch to move ahead with any required changes.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+specific TCK.
+
+== Branches
+
+This project uses branches to target running on specific versions of https://github.com/wildfly/wildfly[WildFly]. For
+example the 31.x branch will be used to run against the 31.x branch of WildFly. 
+
+=== Updating for branching
+
+Once the WildFly branch is crated, a new branch should be created before any new PR's are merged. The branch name
```
Comment (scottmarlow) [inline README.adoc:13]: With around 4 major WildFly releases per year we can expect that over a long period of time we could end up with a lot of branches.  Having said that, this sounds like a good approach.

File: README.adoc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+must match the branch name WildFly used.
```
Comment (scottmarlow) [inline README.adoc:14]: :+1:

File: README.adoc (hunk 1 above)
Comment (jamezp) [reply README.adoc:13]: > With around 4 major WildFly releases per year we can expect that over a long period of time we could end up with a lot of branches. Having said that, this sounds like a good approach.

We do this in wildfly/wildfly itself which is why I thought we should do this. After the next micro release we cold always create a tag and remove the branch.

> 
> Just found a typo, `crated` should be `created`.

Nice catch, I'll fix it!

Comment (jamezp) [reply README.adoc]: Typo fixed, thank you.

**Review by scottmarlow — APPROVED** [review]

**Pull request merged**